packages/jaeger-ui/src/components/TracePage/TraceStatistics/tableValues.tsx

```diff
@@ -106,125 +36,25 @@ function calculateContent(trace: Trace, span: Span, allSpans: Span[], resultValu
   }
   // selfTime
   let tempSelf = 0;
-  let longerAsParent = false;
-  let kinderSchneiden = false;
-  let allOverlay = [];
-  const longerAsParentSpan = [];
   if (span.hasChildren) {
-    const allChildren = [] as any;
-    for (let i = 0; i < allSpans.length; i++) {
-      // i am a child?
-      if (allSpans[i].references.length === 1) {
-        if (span.spanID === allSpans[i].references[0].spanID) {
-          allChildren.push(allSpans[i]);
-        }
-      }
-    }
-    // i only have one child
-    if (allChildren.length === 1) {
-      if (
-        span.relativeStartTime + span.duration >=
-        allChildren[0].relativeStartTime + allChildren[0].duration
-      ) {
-        tempSelf = span.duration - allChildren[0].duration;
-      } else {
-        tempSelf = allChildren[0].relativeStartTime - span.relativeStartTime;
-      }
-    } else {
-      // is the child longer as parent
-      for (let i = 0; i < allChildren.length; i++) {
-        if (
-          span.duration + span.relativeStartTime <
-          allChildren[i].duration + allChildren[i].relativeStartTime
-        ) {
-          longerAsParent = true;
-          longerAsParentSpan.push(allChildren[i]);
-        }
-      }
-      // Do the children overlap?
-      for (let i = 0; i < allChildren.length; i++) {
-        for (let j = 0; j < allChildren.length; j++) {
-          // aren't they the same kids?
-          if (allChildren[i].spanID !== allChildren[j].spanID) {
-            // if yes the children cut themselves or lie into each other
-            if (
-              allChildren[i].relativeStartTime <= allChildren[j].relativeStartTime &&
-              allChildren[i].relativeStartTime + allChildren[i].duration >= allChildren[j].relativeStartTime
-            ) {
-              kinderSchneiden = true;
-              allOverlay.push(allChildren[i]);
-              allOverlay.push(allChildren[j]);
-            }
-          }
-        }
+    const spanDRange = new DRange(span.startTime, span.startTime + span.duration);
+    const childrenDrange = new DRange();
```

Comment:
I don't know why you need to complicate this with a separate range. I think this should work just fine:
```
const spanRange = new DRange(span.startTime, span.startTime + span.duration);
const children = ... // find real children
children.forEach(child => {
    spanRange.subtract(child.startTime, child.startTime + child.duration);
});
selfTime = spanRange.map(r => r.high - r.low).reduce((sum, v) => sum + v, 0);
```

One thing you need to be careful is off-by-one errors. From my understanding drange builds intervals as whole numbers, it does not understand inclusive/exclusive boundaries. For example, when you define `DRange(start, start+duration)`, you're really defining half-open interval `[start, end)` with `selfTime==end-start`, but it's not how drange works. For example:
```
var drange = require("drange");
let r = new drange(1, 10);
r.add(20, 30);
console.log(r.toString()); // "[ 1-10, 20-30 ]"
r.subtract(8, 25);
console.log(r.toString()); // "[ 1-7, 26-30 ]"
```
Notice that even though I logically subtracted `[8, 25)`, the first remaining segment is `1-7`. even though we really want `[1, 8)` half-open interval. So you can compute ST=7-1=6 or ST=8-1=7 - off-by-one (7 is the correct one in this situation).

Reply:
https://github.com/jaegertracing/jaeger-ui/blob/3db728fcac1795c003c37f77118cc4ed05154513/packages/jaeger-ui/src/components/TracePage/TraceStatistics/tableValues.tsx#L49-L52
I am very bad in writing comments 😅

> range builds intervals as whole numbers, it does not understand inclusive/exclusive boundaries.

As you explained above

`let r = new drange(1, 10)`
means r = {1,2,3,4,5,6,7,8,9,10} which we can consider as `1<= t <= 10`

`r.subtract(2, 5)`

means r - {2,3,4,5} = {1,6,7,8,9,10} which we can consider as `t=1 and 5<=t<=10`  (Which is wrong)

but we do +1 and -1 to avoid those errors
`r.subtract(2+1, 5-1)`

means r - {3,4} = {1,2,5,6,7,8,9,10} which we can consider as `1<=t<=2 and 5<=t<=10` (Which is correct)